### PR TITLE
Automation project use Page Factory pattern

### PR DESCRIPTION
**Pull request by VictoriaBadenko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -42,6 +42,11 @@
             <artifactId>selenium-chrome-driver</artifactId>
             <version>4.12.1</version>
         </dependency>
+        <dependency>
+            <groupId>io.github.bonigarcia</groupId>
+            <artifactId>webdrivermanager</artifactId>
+            <version>5.5.3</version>
+        </dependency>
```
Comment (LikBez123) [inline pom.xml]: same here for `webdrivermanager`

File: src/main/java/yandex/mail/driverManager/DriverManager.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+public class DriverManager {
+    private static ThreadLocal<WebDriver> driver = new ThreadLocal<>();
+
+    public DriverManager() {
+    }
+
+    public static WebDriver getDriverInstance() {
+        if (driver.get() == null) {
+            WebDriverManager.chromedriver().setup();
```
Comment (LikBez123) [inline src/main/java/yandex/mail/driverManager/DriverManager.java]: same

File: src/main/java/yandex/mail/driverManager/DriverManager.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package yandex.mail.driverManager;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class DriverManager {
+    private static ThreadLocal<WebDriver> driver = new ThreadLocal<>();
```
Comment (LikBez123) [inline src/main/java/yandex/mail/driverManager/DriverManager.java]: final

File: src/main/java/yandex/mail/pages/InboxPage.java [hunk 4]
```diff
@@ -1,10 +1,26 @@
 package yandex.mail.pages;
 
-import static yandex.mail.Locators.MAIL_CONTAINER;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
 
 public class InboxPage extends BasePage {
+    @FindBy(css = ".mail-Layout-Container")
+    WebElement mailContainer;
```
Comment (LikBez123) [inline src/main/java/yandex/mail/pages/InboxPage.java]: add access modifiers

File: src/main/java/yandex/mail/pages/LoginPage.java [hunk 5]
```diff
@@ -1,41 +1,50 @@
...
-import org.openqa.selenium.support.ui.ExpectedConditions;
-import org.openqa.selenium.support.ui.FluentWait;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
 
-import java.time.Duration;
-import java.util.NoSuchElementException;
+public class LoginPage extends BasePage {
+    @FindBy(css = "#passp-field-login")
+    WebElement usernameInput;
```
Comment (LikBez123) [inline src/main/java/yandex/mail/pages/LoginPage.java]: private

File: src/test/java/seleniumTest/BaseTest.java [hunk 6]
```diff
@@ -1,25 +1,25 @@
...
+        driver = DriverManager.getDriverInstance();
         driver.manage().window().maximize();
         driver.manage().timeouts().implicitlyWait(ofSeconds(15));
-        BasePage.setDriver(driver);
     }
 
-    @AfterMethod(alwaysRun = true)
+    @AfterEach
     public void tearDown() {
         driver.quit();
```
Comment (LikBez123) [inline src/test/java/seleniumTest/BaseTest.java]: this solution does not work for the test suite. Use your singleton class

File: src/test/java/seleniumTest/BaseTest.java [hunk 7]
```diff
@@ -1,25 +1,25 @@
 package seleniumTest;
 
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.chrome.ChromeDriver;
-import org.testng.annotations.AfterMethod;
-import org.testng.annotations.BeforeMethod;
+import yandex.mail.driverManager.DriverManager;
 import yandex.mail.pages.BasePage;
```
Comment (LikBez123) [inline src/test/java/seleniumTest/BaseTest.java]: remove unused imports for the entire project

**Review by LikBez123 — CHANGES_REQUESTED** [review]

**Review by LikBez123 — APPROVED** [review]

**Review by victortylikov — APPROVED** [review]

**Pull request merged**